crates/cli/src/main.rs

```diff
@@ -407,3 +424,28 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    static TELEMETRY: OnceCell<Telemetry> = OnceCell::const_new();
+    let telemetry = TELEMETRY
+        .get_or_init(|| async {
+            Telemetry::new(
+                env!("CARGO_PKG_NAME"),
+                env!("CARGO_PKG_VERSION"),
+                TELEMETRY_CONFIG_NAME,
+                Some(POSTHOG_API_KEY.into()),
+                None,
+                None,
+            )
+            .await
+            .expect("Failed to initialize telemetry")
+        })
+        .await;
```

Comment:
If a static instance is used, it's uncommon to actually pass it as an argument. Instead, probably a better approach would be to create a new crate `anvil_zksync_telemetry`, where you will have something like:

```rust
static TELEMETRY: OnceCell<Telemetry> = OnceCell:new();

pub async fn init_telemetry(app_name: &str, app_version: &str) -> anyhow::Result<()> {
  let telemetry = ...;
  TELEMETRY.set(telemetry).map_err(|_| anyhow::format_err!("Telemetry is already set")
}

pub fn get_telemetry() -> Option<&'static Telemetry> {
  TELEMETRY.get()
}
```

Then we call `init_telemetry` once, and then call `get_telemetry()` whenever we need access to the client.

Some other telemetry-specific stuff (like config name and API key) can also be moved there.

Reply:
Yeah, but I wanted to do that in the lib itself, so that we don't have to do it in every project we use it. Thought this could be refactored later, but it's actually not a big deal to do it now. Will do.

Reply:
Done